hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1876,13 +1876,41 @@ func taintsToJSON(taints []hyperv1.Taint) (string, error) {
 	return string(taintsInJSON), nil
 }
 
-func (r *NodePoolReconciler) reconcileMachineHealthCheck(mhc *capiv1.MachineHealthCheck,
+func (r *NodePoolReconciler) reconcileMachineHealthCheck(ctx context.Context,
+	mhc *capiv1.MachineHealthCheck,
 	nodePool *hyperv1.NodePool,
+	hc *hyperv1.HostedCluster,
 	CAPIClusterName string) error {
+
+	log := ctrl.LoggerFrom(ctx)
+
 	// Opinionated spec based on
 	// https://github.com/openshift/managed-cluster-config/blob/14d4255ec75dc263ffd3d897dfccc725cb2b7072/deploy/osd-machine-api/011-machine-api.srep-worker-healthcheck.MachineHealthCheck.yaml
 	// TODO (alberto): possibly expose this config at the nodePool API.
 	maxUnhealthy := intstr.FromInt(2)
+	timeOut := 8 * time.Minute
+
+	maxUnhealthyOverride := nodePool.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
+	if maxUnhealthyOverride == "" {
+		maxUnhealthyOverride = hc.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
+	}
+	if maxUnhealthyOverride != "" {
+		maxUnhealthy = intstr.FromString(maxUnhealthyOverride)
```

Comment:
MHC will transparently fail if this value is not legit?

Reply:
true, let me see if this can be validated first